src/pybind/mgr/dashboard/controllers/rbd.py

```diff
@@ -1,136 +1,421 @@
 # -*- coding: utf-8 -*-
+# pylint: disable=too-many-arguments,too-many-locals,unused-argument
 from __future__ import absolute_import
 
 import math
 import cherrypy
+import rados
 import rbd
 
-from . import ApiController, AuthRequired, RESTController
+from . import ApiController, AuthRequired, RESTController, Task
 from .. import mgr
+from ..services.ceph_service import CephService
 from ..tools import ViewCache
 
 
+# pylint: disable=inconsistent-return-statements
+def _rbd_exception_handler(ex):
+    if isinstance(ex, rbd.OSError):
+        return {'status': 409, 'detail': str(ex), 'errno': ex.errno,
+                'component': 'rbd'}
+    elif isinstance(ex, rados.OSError):
+        return {'status': 409, 'detail': str(ex), 'errno': ex.errno,
+                'component': 'rados'}
+    raise ex
+
+
+def RbdTask(name, metadata, wait_for):
+    return Task("rbd/{}".format(name), metadata, wait_for,
+                _rbd_exception_handler)
+
+
+def _rbd_call(pool_name, func, *args, **kwargs):
+    with mgr.rados.open_ioctx(pool_name) as ioctx:
+        func(ioctx, *args, **kwargs)
+
+
+def _rbd_image_call(pool_name, image_name, func, *args, **kwargs):
+    def _ioctx_func(ioctx, image_name, func, *args, **kwargs):
+        with rbd.Image(ioctx, image_name) as img:
+            func(ioctx, img, *args, **kwargs)
+
+    return _rbd_call(pool_name, _ioctx_func, image_name, func, *args, **kwargs)
+
+
+RBD_FEATURES_NAME_MAPPING = {
+    rbd.RBD_FEATURE_LAYERING: "layering",
+    rbd.RBD_FEATURE_STRIPINGV2: "striping",
+    rbd.RBD_FEATURE_EXCLUSIVE_LOCK: "exclusive-lock",
+    rbd.RBD_FEATURE_OBJECT_MAP: "object-map",
+    rbd.RBD_FEATURE_FAST_DIFF: "fast-diff",
+    rbd.RBD_FEATURE_DEEP_FLATTEN: "deep-flatten",
+    rbd.RBD_FEATURE_JOURNALING: "journaling",
+    rbd.RBD_FEATURE_DATA_POOL: "data-pool",
+    rbd.RBD_FEATURE_OPERATIONS: "operations",
+}
+
+def _format_bitmask(features):
```

Comment:
The linter rightfully complained here: `./controllers/rbd.py:57:1: E302 expected 2 blank lines, found 1`